medusa/name_parser/guessit_parser.py

```diff
@@ -75,8 +75,8 @@ def guessit(name, options=None):
     return result
 
 
-def get_expected_titles(show_list):
-    """Return expected titles to be used by guessit.
+def gen_expected_titles(show_list):
```

Comment:
This function is very expensive. For each title it will go through this method. Which includes a union, regex and three nested loops. Changing it to an iterator could help depending how it's used. But I could not understand how it's used. Maybe you know more about this, as it's more in the guessit domain.

Reply:
I feal that the list should only be build once per search, but then we have to change it back from generator to iterator.